docs/reference/r-requirements.md

```diff
@@ -1,4 +1,13 @@
-# Requirements
+## Juju version
+
+The charm supports both [Juju 2.9 LTS](https://github.com/juju/juju/releases) and [Juju 3.1](https://github.com/juju/juju/releases). However, we always advise new deployment to be carried out on Juju 3. 
+
+The minimum supported Juju versions are:
+
+* 2.9.32+ (no tests made for older versions).
+* 3.1.6+ (due to issues with Juju secrets in previous versions, see [#1](https://bugs.launchpad.net/juju/+bug/2029285) and [#2](https://bugs.launchpad.net/juju/+bug/2029282))
```

Comment:
**todo:** Same as other comments here, we should drop this.

If we published `edge` to `stable` tomorrow (which is what docs should be representing IMO), then we would ONLY support `3.1.6+`. I think we should just state that, and remove any reference to 2.9.

Reply:
I have revised the sentence a bit, to make sure we clarify that 2.9 is now deprecated and support in 2.9 may be discontinued soon